ProjectorTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;

```

Comment:
This was good practise for writing a test class but for an abstract class, it doesn't actually test anything so you can just delete this class

Reply:
When I created a unit test it created all this automatically. And as you add your functionalities to the class you add it into the test class. I only have a abstract classes. Where are the nonabstract classes I need to test?